pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -300,10 +310,12 @@ func managePod_v311_00_to_latest(ctx context.Context, configMapsGetter corev1cli
 	configMap.Data["forceRedeploymentReason"] = operatorSpec.ForceRedeploymentReason
 	configMap.Data["version"] = version.Get().String()
 	appliedConfigMap, changed, err := resourceapply.ApplyConfigMap(ctx, configMapsGetter, recorder, configMap)
+	deprecatedPolicy := false
 	if changed && len(config.Spec.Policy.Name) > 0 {
+		deprecatedPolicy = true
```

Comment:
**Observation**: The scheduler operator registers the event handler for `config.openshift.io/v1/scheduler`. When `spec.Policy.Name` field in the `scheduler/cluster` object is updated, the `sync` method is triggered and `managePod_v311_00_to_latest` is invoked.
  - If the `managePod_v311_00_to_latest` function returns an error, the operator condition `TargetConfigControllerDegraded` is set to true.
  - If the `managePod_v311_00_to_latest` function does not return an error, the `changed` is set to true and the `deprecatedPolicy = len(config.Spec.Policy.Name) > 0`.

So when the `spec.Policy.Name` field is non-empty, either
- `condition/TargetConfigControllerDegraded=true`
- or `condition/OperatorStatusTypeUpgradeable=false`.

Assuming the cluster upgrade requires all operators to have
- `condition/TargetConfigControllerDegraded=false`
- and `condition/OperatorStatusTypeUpgradeable=true`

this change will stop the upgrade from happening when the policy API is in use (assuming the operator status will get updated on time).

Reply:
Are you saying that something with `spec.Policy.Name` would cause `managePod` to return an error? Or just observing the possible cases where `managePod` returns an error *before* we actually set `deprecatedPolicy = true`?

Reply:
I wanted to verify the operator will properly report a condition disallowing the cluster upgrade by checking possible cases of when the `spec.Policy.Name` is updated. I did not find any suspicious line of code which would claim otherwise.